src/sagemaker_xgboost_container/serving.py

```diff
@@ -43,6 +43,13 @@ def is_multi_model():
     return os.environ.get("SAGEMAKER_MULTI_MODEL")
 
 
+def set_default_env():
+    env_default_dict = {"OMP_NUM_THREADS": "1"}
+    for key, value in env_default_dict.items():
+        if os.getenv(key) is None:
+            os.environ[key] = value
```

Comment:
```suggestion
        user_specified_value = os.getenv(key)
        if user_specific_value is not None:
            os.environ[key] = user_specific_value
```

Reply:
Thanks for the reply but I am a little bit confused right here. So this function basically was to set ENV variables with default values if users haven't specified them.  

What do you mean by user_specific_value? (Correct me if I am wrong) From my understanding, OMP_NUM_THREADS = 1 should be a default setting for all inference tasks. That's why I use set_default_env for the function name. 

I will set "1" to a constant rather than hard coding

Reply:
@awsbmillare Perhaps a better function name if that's the intended behavior then is something like `set_unspecified_env_value_defaults`. 

In the original code, if the keys in `env_default_dict` aren't already defined in the environment then they'll be set to the default values. But if they are already defined (and many libraries set `OMP_NUM_THREADS`) then nothing happens - just want to check that this is intended and will work in all execution paths.

Reply:
Can users specify environment variables for the container? If so, does this change allow them to override and set a custom value for `OMP_NUM_THREADS`?

Reply:
I think its more we are grabbing the current env vars and putting it in the dict (to make it easy to walk through). If OMP_NUM_THREADs is undefined, then we want to set it to a default of 1. If it does have a setting, we take that as the customer wants to specify this and we let that be the value observed. This way the customer can still tune the thread count for optimal performance. 1 is likely not always optimal and depends on different factors.